src/screens/ArtisanHub/ArtisanDetail.js

```diff
@@ -54,209 +52,254 @@ class ArtisanDetail extends Component {
     this.navigateToEditArtisan = this.navigateToEditArtisan.bind(this)
     this.fetchProducts = this.fetchProducts.bind(this)
     this.sortedProducts = this.sortedProducts.bind(this)
-    
   }
 
-  fetchProducts() {
-    const { uid } = this.props
-    this.setState({fetchingProducts: true})
-    this.props.fetchProducts(uid).then(() => {
-      this.setState({fetchingProducts: false, currentProducts: this.props.Products})
-    })
+  componentDidMount() {
+    this.fetchProducts()
+  }
+
+  componentDidUpdate() {
+    LayoutAnimation.easeInEaseOut()
   }
 
   onCancel() {
     this.setState({ adding: false })
   }
 
+  deletePressed() {
+    this.setState({ adding: true })
+    this.props.deleteArtisan(this.props.Artisans, this.props.uid).then(() => {
+      this.setState({ adding: false })
+      this.props.navigation.navigate('ArtisanList')
+    })
+  }
+
+  fetchProducts() {
+    const { uid } = this.props
+    this.setState({ fetchingProducts: true })
+    this.props.fetchProducts(uid).then(() => {
+      this.setState({
+        fetchingProducts: false,
+        currentProducts: this.props.Products
+      })
+    })
+  }
+
   navigateToEditArtisan() {
```

Comment:
Not sure if this deserves its own method. Use the spread operator to pass the object as variables.